src/rod/utils/frame_convention.py

```diff
@@ -288,33 +292,37 @@ def find_parent_link_of_frame(frame: rod.Frame, model: rod.Model) -> str:
 
     assert isinstance(frame, rod.Frame)
 
-    if frame.attached_to in links_dict:
-        parent = links_dict[frame.attached_to]
+    match frame.attached_to:
+        case anchor if anchor in links_dict:
+            parent = links_dict[frame.attached_to]
 
-    elif frame.attached_to in frames_dict:
-        parent = frames_dict[frame.attached_to]
+        case anchor if anchor in frames_dict:
+            parent = frames_dict[frame.attached_to]
 
-    elif frame.attached in {model.name, "__model__"}:
-        return model.get_canonical_link()
+        case frame if frame.attached in {model.name, "__model__"}:
```

Comment:
Somehow the fix of https://github.com/ami-iit/rod/pull/31#issuecomment-2117622796 got lost:

```suggestion
        case frame if frame.attached_to in {model.name, "__model__"}:
```

Reply:
Done in https://github.com/ami-iit/rod/pull/31/commits/77a5f47aae238a396995ca226fe1eee56f32fcb3